tests/test_regressions.py

```diff
@@ -412,6 +412,9 @@ def test_format_invalid_where_clause():
     formatted = sqlparse.format('where, foo', reindent=True)
     assert formatted == 'where, foo'
 
+def test_cte_named_final():
+    p = sqlparse.parse('with final as (select 1) select final')[0]
+    assert p.get_type() == 'SELECT'
```

Comment:
This used to report `'UNKNOWN'` before this fix, which is what was causing issues for our users.